test/CadlRanchProjects/client/structure/two-operation-group/src/Generated/TwoOperationGroupClient.cs

```diff
@@ -0,0 +1,79 @@
+        /// <param name="endpoint"> Need to be set as 'http://localhost:3000' in client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> is null. </exception>
+        public TwoOperationGroupClient(Uri endpoint) : this(endpoint, new TwoOperationGroupClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of TwoOperationGroupClient. </summary>
+        /// <param name="endpoint"> Need to be set as 'http://localhost:3000' in client. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> is null. </exception>
+        public TwoOperationGroupClient(Uri endpoint, TwoOperationGroupClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            options ??= new TwoOperationGroupClientOptions();
+
+            ClientDiagnostics = new ClientDiagnostics(options, true);
+            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), Array.Empty<HttpPipelinePolicy>(), new ResponseClassifier());
+            _endpoint = endpoint;
+        }
+
+        /// <summary> Initializes a new instance of Group1. </summary>
+        /// <param name="client"> Need to be set as 'default', 'multi-client', 'renamed-operation', 'two-operation-group' in client. Allowed values: "default" | "multi-client" | "renamed-operation" | "two-operation-group". </param>
+        /// <param name="apiVersion"> The String to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="client"/> or <paramref name="apiVersion"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="client"/> is an empty string, and was expected to be non-empty. </exception>
+        public virtual Group1 GetGroup1Client(string client, string apiVersion = "1.0.0")
+        {
+            Argument.AssertNotNullOrEmpty(client, nameof(client));
+            Argument.AssertNotNull(apiVersion, nameof(apiVersion));
+
+            return new Group1(ClientDiagnostics, _pipeline, _endpoint, client, apiVersion);
+        }
+
+        /// <summary> Initializes a new instance of Group2. </summary>
+        /// <param name="client"> Need to be set as 'default', 'multi-client', 'renamed-operation', 'two-operation-group' in client. Allowed values: "default" | "multi-client" | "renamed-operation" | "two-operation-group". </param>
+        /// <param name="apiVersion"> The String to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="client"/> or <paramref name="apiVersion"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="client"/> is an empty string, and was expected to be non-empty. </exception>
+        public virtual Group2 GetGroup2Client(string client, string apiVersion = "1.0.0")
```

Comment:
do you know why we need to specify api version here?  Is it by design?  Seems this is the same defined in clientOptions.

Reply:
Do we allow other values?

Reply:
This is a method to get subclient. it will add apiVersion with a default value. Customer can set other value. If the api-version is not supported, the api call will report error.